### PR TITLE
Part 1 Scissors Paper Stone ✂️ 📄 🪨 Project

### PR DESCRIPTION
**Pull request by crystalmoh**
Please fill out the survey before submitting the pull request. Thanks!

🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀

How many hours did you spend on this assignment? 
2 hours

Please fill in one error and/or error message you received while working on this assignment.
git push 
"error: failed to push some refs to"

What part of the assignment did you spend the most time on?
Writing pseudo codes and defining the logics. 

Comfort Level (1-5):
4
Completeness Level (1-5):
4
What did you think of this deliverable?
I liked the option of the reverse prank game, it enforces versatility

Is there anything in this code that you feel pleased about?
Yes, creating nicer formatting in the output.

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -1,4 +1,140 @@
...
+    // switch game mode to Scissors Paper Stone once name is given
+    currentGameMode = "scissors paper stone game";
+    myOutputValue =
+      "hello " +
+      input +
+      ", you are now playing scissors paper stone! Please only choose the following:<br>✂️scissors<br>🧻paper<br><b>or</b> 🪨stone";
+    return myOutputValue;
+  } else if (currentGameMode == "scissors paper stone game") {
+    // scissors paper stone logic added here
+    var computerRandom = generateRandomOutcome();
```
Comment (zephaniahong) [inline script.js:21]: Good use of a helper function

File: script.js [hunk 2]
```diff
@@ -1,4 +1,140 @@
...
+    var myOutputValue = `${userName}, you chose ${input},<br>the computer chose ${computerRandom}`;
+    console.log("Computer Chose: " + computerRandom);
+    console.log(`User Chose: ${input}`);
+    if (
+      (input == "scissors" && computerRandom == "paper") ||
+      (input == "paper" && computerRandom == "stone") ||
+      (input == "stone" && computerRandom == "scissors")
+    ) {
```
Comment (zephaniahong) [inline script.js:29]: Consider abstracting this logic to a helper function, such as checkWinner(userInput, computerInput).

File: script.js [hunk 3]
```diff
@@ -1,4 +1,140 @@
...
+      totalGamesPlayed = totalGamesPlayed + 1;
+      totalGamesWon = totalGamesWon + 1;
+      if (totalGamesWon / totalGamesPlayed >= 0.5) {
+        return `🎉 Congratulations! <br> ${userName}, you won! 🤗<br>You chose ${input} and the computer chose ${computerRandom}! <br> You have been winning ${totalGamesWon} out of ${totalGamesPlayed} rounds! It's looking good!`;
+      } else {
+        return `🎉 Congratulations! <br> ${userName}, you won! 🤗<br>You chose ${input} and the computer chose ${computerRandom}! <br> You have been winning ${totalGamesWon} out of ${totalGamesPlayed} rounds! Try harder!`;
```
Comment (zephaniahong) [inline script.js:35]: Similarly over here, you can have a helper function called message(totalGamesWon, totalGamesPlayed) that returns the message

File: script.js [hunk 4]
```diff
@@ -1,4 +1,140 @@
...
+    }
+
+    if (
+      input != "scissors" &&
+      input != "paper" &&
+      input != "stone" &&
+      input != "reversed scissors" &&
+      input != "reversed paper" &&
+      input != "reversed stone"
+    ) {
```
Comment (zephaniahong) [inline script.js:70]: This logic could be abstracted into a helper function called checkValid(input)

**Review by zephaniahong — COMMENTED** [review]
I can see that you have a good understanding of booleans and conditional statements. One thing that could be improved on would be using more helper functions to abstract logic out of the main function to make your main function a lot cleaner. Overall, a very good attempt!

File: script.js (hunk 2 above)
Comment (crystalmoh) [reply script.js:29]: thank you Zeph!

**Pull request closed without merging**